awswrangler/distributed/ray/datasources/arrow_parquet_datasource.py

```diff
@@ -19,9 +19,9 @@
 from pyarrow.dataset import ParquetFileFragment
 from pyarrow.lib import Schema
 from ray import cloudpickle
+from ray.data._internal.output_buffer import BlockOutputBuffer
 from ray.data.block import Block, BlockAccessor
 from ray.data.context import DatasetContext
-from ray.data.dataset import BlockOutputBuffer  # type: ignore[attr-defined]
```

Comment:
I have checked and this import change is backward compatible across all ray 2.0+ versions